convert2rhel_insights_tasks/main.py

```diff
@@ -862,32 +890,33 @@ def main():
         check_convert2rhel_inhibitors_before_run()
         stdout, returncode = run_convert2rhel()
         execution_successful = returncode == 0
-        rollback_errors = check_for_inhibitors_in_rollback()
 
-        # Check if there are any inhibitors in the rollback logging. This is
-        # necessary in the case where the analysis was done successfully, but
-        # there was an error in the rollback log.
-        if rollback_errors:
-            raise ProcessError(
-                message=(
-                    "A rollback of changes performed by convert2rhel failed. The system is in an undefined state. "
-                    "Recover the system from a backup or contact Red Hat support."
-                ),
-                report=(
-                    "\nFor details, refer to the convert2rhel log file on the host at "
-                    "/var/log/convert2rhel/convert2rhel.log. Relevant lines from log file: \n%s\n"
-                )
-                % rollback_errors,
-            )
-
-        # Returncode other than 0 can happen in two states in analysis mode:
+        # Returncode other than 0 can happen in three states in analysis mode:
         #  1. In case there is another instance of convert2rhel running
         #  2. In case of KeyboardInterrupt, SystemExit (misplaced by mistaked),
         #     Exception not catched before.
+        #  3. There was an error during rollback. This result in return code 1.
         # In any case, we should treat this as separate and give it higher
         # priority. In case the returncode was non zero, we don't care about
         # the rest and we should jump to the exception handling immediatly
         if not execution_successful:
+            rollback_errors = get_rollback_failures(returncode)
```

Comment:
we know returncode is not 0 here thanks to `execution_successful` var, couldn't we just instead of changing two functions to take returncode param update the `execution_successful` to also take into account None and then use it for rollback and for gathering report data? Or what is the point of `execution_successful` condition now? I think we can work either work with just returncode or the `execution_successful` var.

Reply:
My motivation for adding the `returncode` into the functions was to not increase the complexity of the `main()` function by adding some conditions there - I put the parameter `returncode` and let the evaluation run inside the functions. 

I was also thinking about adding the handling to the `main()`, but I didn't like the increased complexity - I know, it would be just a few lines, but still... It's something.

And for the `execution_successfull` I didn't want to change the logic of the script much  - this is a part you already had and is working. But this is something I can change, but still: we still need to check if the `returncode == 0` in two places, so maybe it's useful?

Because the value of the `returncode` can be:
- `None` if the conversion hasn't run 
- `0` success
- `1` rollback fail or **WIP**: need to check, after looking into the code of c2r there are some things which goes against each other. This is also related to your next question
- `2` inhibitor

Reply:
All I wanted to point out is that for me the `execution_successful` is easier to read, logic with return code makes sense, but it's harder to decode so if it would be hidden behind some name and then used in the main function and on other places it would be nicer.

If all the cases are passing tests I am fine with it :)

Reply:
I think we can proceed with this for now, and we can revisit it in the future.